syntax = "proto3";
package monitor.proto;




import "google/protobuf/empty.proto";
import "cpu_load.proto";
import "cpu_softirq.proto";
import "cpu_stat.proto";
import "mem_info.proto";
import "net_info.proto";



message MonitorInfo
{
    string name = 1;
    repeated CpuSoftirqs soft_irq = 2;
    CpuLoad cpu_load = 3; 
    repeated CpuStat cpu_stat = 4;
    MemInfo mem_info = 5;
    NetInfo net_info = 6;
}



service RpcManager
{
    rpc SetMonitorInfo(MonitorInfo)returns(google.protobuf.Empty)
    {}
    rpc GetMonitorInfo(google.protobuf.Empty)returns(MonitorInfo)
    {}
}